### PR TITLE
[enocean] Added support for SIG-Messages (battery level indication)

### LINKED ISSUE #6520: [enocean] Support battery level messages
Certain EnOcean devices send so called Signal messages to indicate their battery level. These messages should be supported by this binding, too.

### PR DESCRIPTION
**Pull request by fruggy83**
* Added support for SIG-Messages
 * Added new EEP D0-06 to handle battery level SIG-Messages
 * Added a second receiving EEP for D0-06 to all sensors which are likely battery powered

Fixes #6520

Signed-off-by: Daniel Weber <uni@fruggy.de>

### CODE REVIEW

**Comment by fruggy83** [comment]
This PR will conflict with #6513 so a rebase is needed after merging one of these branches into master.

**Comment by fruggy83** [comment]
Rebased on 2.5.x and solved conflicts

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/Base/_SIGMessage.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+package org.openhab.binding.enocean.internal.eep.Base;
+
+import org.openhab.binding.enocean.internal.eep.EEP;
+import org.openhab.binding.enocean.internal.messages.ERP1Message;
+
+/**
+ *
+ * @author Daniel Weber - Initial contribution
+ */
+public abstract class _SIGMessage extends EEP {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/Base/_SIGMessage.java]: According to our codestyle class names have to match '^[A-Z][a-zA-Z0-9]*$'. Is there any reason why this cannot be applied here?

File: bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml [hunk 2]
```diff
@@ -18,13 +18,15 @@
 				<description>EnOceanId of device this thing belongs to</description>
 				<required>true</required>
 			</parameter>
-			<parameter name="receivingEEPId" type="text">
+			<parameter name="receivingEEPId" type="text" multiple="true">
```
Comment (J-N-K) [inline bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: Buts this also allows to select `A5_06_01` and `A5_06_01_ELTAKO` which is probably not what you intended. What happens if the user does that? (also applies below)

File: bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/Base/_SIGMessage.java (hunk 1 above)
Comment (fruggy83) [reply bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/Base/_SIGMessage.java]: Sorry I was not aware of this codestyle. The most common EnOcean messages are based on RPS, 1BS, 4BS and VLD messages. So I have choosen _RPSMessage, _1BSMessage, _4BSMessage and _VLDMessage for my base classes as it is not allowed to start a class name with a numeric value. As this _SIGMessage class is also a base class for certain messages I just used the same naming schema. 
For auto completion reasons in my IDE I would like to keep the differentiating part in first place of the class name.

File: bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml (hunk 2 above)
Comment (fruggy83) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: As the choosen EEPs define the channels of a thing I added the D0_06_00 EEP to the receivingEEP too. Later during validation of the thing configuration I am checking if the user has choosen max one EEP of each EEP family (RPS, 1BS, 4BS, VLD, SIG).
So it is true that I do not want the user to select ```A5_06_01``` and ```A5_06_01_ELTAKO```, but adding a new config param for each multi EEP receiving case would have made the implementation a little bit more complicated.

Comment (J-N-K) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: Can you point me to the code that validates this`?

Comment (fruggy83) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: Yes for sure, validation is done in the EnOceanBaseSensorHandler, function [validateConfig()](https://github.com/openhab/openhab-addons/blob/469a6ae822ea7987d3bb2c438a99c657510a8678/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBaseSensorHandler.java#L82) Line 95

Comment (J-N-K) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: I prepared a PR against your fork. WDYT?

Comment (fruggy83) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: @J-N-K  Thanks for the proposal. I really like your approach. I will accept your PR, rework the thing xmls and update the PR. Thanks a lot. By the way, I am always learning something new from your code, was not aware of the shorthand function pointer/lambda in line 158 👍 .

Comment (fruggy83) [reply bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/LightSensor.xml]: @J-N-K thanks again for your proposal. I updated my PR accordingly.

**Comment by J-N-K** [comment]
@fruggy83 Can you adress the last two open points?

**Comment by fruggy83** [comment]
@J-N-K sorry missed the review result. I will try to fix these points tomorrow. Which of the two open PR do you want to merge first. As @cweitkamp suggests this PR will conflict with #6930 So I would suggest to first merge this PR, fix the open points in #6930 and do a rebase afterwards.

**Comment by J-N-K** [comment]
I would suggest to merge this one first

**Comment by fruggy83** [comment]
@cweitkamp , @J-N-K  thanks for your hints. Fixed the last two open points. After merging this PR I would rebase #6930 and fix its issues. Thanks a lot 👍

**Pull request merged**